src/util/text-utils.cc

```diff
@@ -188,67 +190,77 @@ inline bool is_nan_text(const std::string &in, const std::string &prefix) {
   return true;
 }
 
-template<typename T>
-bool convert_special_number(const std::string &str, T *out) {
-  if (stricmp(str, "infinity") || stricmp(str, "inf") ||
-      starts_with(str, "1.#INF")) {
-    *out = std::numeric_limits<T>::infinity();
-    return true;
-  } else if (stricmp(str, "-infinity") || stricmp(str, "-inf") ||
-             starts_with(str, "-1.#INF")) {
-    *out = -std::numeric_limits<T>::infinity();
-    return true;
-  } else if (is_nan_text(str, "nan") || starts_with(str, "1.#QNAN")) {
-    *out = std::numeric_limits<T>::quiet_NaN();
-    return true;
-  } else if (is_nan_text(str, "-nan") || starts_with(str, "-1.#QNAN")) {
-    *out = -std::numeric_limits<T>::quiet_NaN();
-    return true;
+template <class T>
+class number_istream
+{
+ public:
+
+  number_istream (std::istream &i) : in_(i) {}
+
+  number_istream & operator >> (T &x) {
+    bool neg = false;
+    char c;
+    if (!in_.good()) return *this;
+    while (isspace(c = in_.peek())) in_.get();
+    if (c == '-') { neg = true; }
+    in_ >> x;
+    if (! in_.fail()) return *this;
+    return parse_on_fail(x, neg);
   }
-  return false;
-}
 
-bool ConvertStringToReal(const std::string &str,
-                         double *out) {
-  const char *this_str = str.c_str();
-  char *end = NULL;
-  errno = 0;
-
-#if defined(_MSC_VER)
-  // TODO: check if the new MSVC already supports it
-  // depending on claims of the C++11 support, it should have
-  if (convert_special_number(str, out))
-    return true;
-#endif  // defined(_MSC_VER)
+ private:
+  std::istream &in_;
+
+  number_istream & parse_on_fail (T &x, bool neg)
+   {
+    std::map<std::string, T> inf_nam_map;
+    // we'll keep just lowercase values.
+    inf_nam_map["inf"] = std::numeric_limits<T>::infinity();
```

Comment:
do you mean nan_map instead of nam_map?  If 'nam' means name, please don't abbreviate like that, google syle guide doesn't like that.